ale_linters/haskell/hlint.vim

```diff
@@ -26,9 +29,21 @@ function! ale_linters#haskell#hlint#Handle(buffer, lines) abort
     return l:output
 endfunction
 
+function! ale_linters#haskell#hlint#GetExecutable(buffer) abort
+    return ale#Var(a:buffer, 'haskell_hlint_executable')
+endfunction
+
+function! ale_linters#haskell#hlint#GetCommand(buffer) abort
+    let l:hlintopts = ' --color=never --json'
+
+    return ale#Escape(ale_linters#haskell#hlint#GetExecutable(a:buffer))
```

Comment:
Try using just `'%e'` here, which will be replaced with the escaped executable name.